Drop custom IPC decoder / encoder for WTF::MachSendRight

cdumez opened the pull request:
#### 5bf2fff545a79c642758d2e15c2d2d41f76cb466
<pre>
Drop custom IPC decoder / encoder for WTF::MachSendRight
<a href="https://bugs.webkit.org/show_bug.cgi?id=268335">https://bugs.webkit.org/show_bug.cgi?id=268335</a>

Reviewed by Alex Christensen.

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Platform/IPC/Connection.cpp:
* Source/WebKit/Platform/IPC/Decoder.h:
(IPC::Decoder::operator&gt;&gt;):
(IPC::Decoder::decode):
* Source/WebKit/Platform/IPC/Encoder.h:
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
(IPC::Connection::receiveSourceEventHandler):
* Source/WebKit/Platform/IPC/darwin/ArgumentCodersDarwin.h: Removed.
* Source/WebKit/Platform/IPC/darwin/ArgumentCodersDarwin.mm: Removed.
* Source/WebKit/Platform/SharedMemory.serialization.in:
* Source/WebKit/Platform/SourcesCocoa.txt:
* Source/WebKit/Scripts/webkit/messages.py:
(argument_coder_headers_for_type):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessageReceiver.cpp:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/273876@main">https://commits.webkit.org/273876@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a8d02d6ffbdc828f189c2ee71e3be34f32e8ae1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36944 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15865 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39205 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39556 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33053 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18395 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12989 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31584 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37505 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13390 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32610 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11681 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/37048 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11702 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40808 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33493 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33252 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37602 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11995 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9804 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35729 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13647 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12375 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4795 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12907 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Platform/IPC/Decoder.h:
```diff
@@ -149,6 +149,12 @@ class Decoder {
         return *this;
     }
 
+    Decoder& operator>>(std::optional<Attachment>& t)
```
Do we really need this, since we have the Decoder::decode overload?

cdumez replied:
Maybe not, I'll check

cdumez replied:
Looks like it wasn't needed, I dropped it.

cdumez committed:
Drop custom IPC decoder / encoder for WTF::MachSendRight
https://bugs.webkit.org/show_bug.cgi?id=268335

Reviewed by Alex Christensen.

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Platform/IPC/Connection.cpp:
* Source/WebKit/Platform/IPC/Decoder.h:
(IPC::Decoder::operator>>):
(IPC::Decoder::decode):
* Source/WebKit/Platform/IPC/Encoder.h:
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
(IPC::Connection::receiveSourceEventHandler):
* Source/WebKit/Platform/IPC/darwin/ArgumentCodersDarwin.h: Removed.
* Source/WebKit/Platform/IPC/darwin/ArgumentCodersDarwin.mm: Removed.
* Source/WebKit/Platform/SharedMemory.serialization.in:
* Source/WebKit/Platform/SourcesCocoa.txt:
* Source/WebKit/Scripts/webkit/messages.py:
(argument_coder_headers_for_type):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessageReceiver.cpp:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/273876@main
